marine-js/npm-package/src/marine_js.js

```diff
@@ -66,6 +66,15 @@ export async function init(module) {
         return cachedTextDecoder.decode(getUint8Memory0().subarray(ptr, ptr + len));
     }
 
+    function addHeapObject(obj) {
+        if (heap_next === heap.length) heap.push(heap.length + 1);
+        const idx = heap_next;
+        heap_next = heap[idx];
```

Comment:
Is this correct? Shouldn't `heap_next` hold the index, not element?

I suggest renaming `heap_next` to `heap_next_idx` or `heap_next_element` to make it clearer.

Reply:
This was generated by wasm-bindgen.  it is a simple data structure.
An array where free slots form a linked list, beginning is in heap_next. They store index of next free slot. If heap_next is heap.length, then there is no free slots. So, constant insert, get, delete by index, plain data structure. But no order guaranteed. Everything looks correct except that this is a redundant copy.